File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/_Tokens/Typography/Typography.swift
import SwiftUI

public struct Typography {
    public let display: Display
    public let subhead: Subhead
    public let body: Body

    public struct Display {
        public let h1: Font
        public let h2: Font
        public let h3: Font
        public let h4: Font
        public let h5: Font
        public let h6: Font
    }

    public struct Subhead {
        public let p1: Font
        public let p2: Font
        public let p3: Font
    }

    public struct Body {
        public let p1: Font
        public let p2: Font
        public let p3: Font
        public let p4: Font
    }
}

File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/_tokens/typography/Typography.kt
@file:Suppress("ktlint:standard:package-name")

package com.perrystreet.woof.designsystem.atomic._tokens.typography

import androidx.compose.runtime.Immutable
import androidx.compose.ui.text.TextStyle

@Immutable
data class Typography(
    val display: Display,
    val subhead: Subhead,
    val body: Body,
) {
    @Immutable
    data class Display(
        val h1: TextStyle,
        val h2: TextStyle,
        val h3: TextStyle,
        val h4: TextStyle,
        val h5: TextStyle,
        val h6: TextStyle,
    )

    @Immutable
    data class Subhead(
        val p1: TextStyle,
        val p2: TextStyle,
        val p3: TextStyle,
    )

    @Immutable
    data class Body(
        val p1: TextStyle,
        val p2: TextStyle,
        val p3: TextStyle,
        val p4: TextStyle,
    )
}
